app/views/samples_reports/nih_challenge.csv.csvbuilder

```diff
@@ -0,0 +1,55 @@
+csv << [
+  'protocol_id',
+  'technology_platform',
+  'vqa_box_id',
+  'sample_group',
+  'target_analyte_type',
+  'target_organism_name',
+  'target_organism_taxonomy_id',
+  'pango_lineage',
+  'who_label',
+  'non_clinical_cutoff_min',
+  'non_clinical_cutoff_unit',
+  'true_positives',
+  'false_positives',
+  'true_negatives',
+  'false_negatives',
+  'non_clinical_sensitivity',
+  'non_clinical_sensitivity_ci_percent',
+  'non_clinical_sensitivity_ci_min',
+  'non_clinical_specificity',
+  'non_clinical_specificity_ci_percent',
+  'non_clinical_specificity_ci_min',
+  'area_under_the_roc_curve',
+  'roc_x_values',
+  'roc_y_values'
+]
+
+sample = @samples_report.samples[0]
+batch = Batch.find_by(batch_number: sample.batch_number)
```

Comment:
You should load `@batch` in the controller directly, or better add a `SamplesReport#batch` method, so the logic to find the batch associated to a SamplesReport is abstracted in the model.

Cherry on the cake: using SQL to find the batch?

```ruby
Batch.joins(:samples).where(samples_report_id: id).take
```

Reply:
Indeed we wanted the non-distractor batch to be taken as the target batch... Thanks to this correction the error is solved :)

Reply:
Corrected by rubocop:

`Rails/FindBy: Use find_by instead of where.take.
    samples_report_samples.joins(:sample).where("samples.distractor IS NULL OR samples.distractor = 'false'").take.sample.batch`